lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionsDsl.kt

```diff
@@ -74,6 +75,37 @@ public object ExpressionsDsl {
   public fun const(list: List<String>): Expression<ListValue<StringValue>> =
     Expression.ofList(listOf("literal", list)).cast()
 
+  /** Creates a literal expression for a list of numbers. */
+  @JvmName("constNumberList")
+  public fun const(list: List<Number>): Expression<VectorValue<FloatValue>> =
+    Expression.ofList(listOf("literal", list)).cast()
+
+  public fun textOffset(x: TextUnit, y: TextUnit): Expression<TextOffsetValue> {
+    require(x.type == y.type) { "x and y must have the same TextUnitType" }
+
+    val reasonablyLargeMultiplier = 1000f
+    val type = x.type
+    val xVal = if (x.isSpecified) x.value else 1f
+    val yVal = if (y.isSpecified) y.value else 1f
+
+    return interpolate(
+        linear(),
+        when (type) {
+            TextUnitType.Sp -> spMultiplierVar
+            TextUnitType.Em -> emMultiplierVar
+            TextUnitType.Unspecified -> unspecifiedValueVar
+            else -> error("Unrecognized TextUnitType: $type")
+          }
+          .use()
+          .cast(),
+        0f to const(listOf(0, 0)),
+        1f to const(listOf(xVal, yVal)),
+        reasonablyLargeMultiplier to
+          const(listOf(xVal * reasonablyLargeMultiplier, yVal * reasonablyLargeMultiplier)),
+      )
+      .cast()
+  }
```

Comment:
if textSize is less than 0.001sp, this would start to break down. Probably not a big deal